Gemfile

```diff
@@ -27,14 +27,15 @@ gem "stimulus-rails"
 # Build JSON APIs with ease [https://github.com/rails/jbuilder]
 gem "jbuilder"
 
+gem "byebug"
 # Use Redis adapter to run Action Cable in production
 # gem "redis", "~> 4.0"
 
 # Use Kredis to get higher-level data types in Redis [https://github.com/rails/kredis]
 # gem "kredis"
 
 # Use Active Model has_secure_password [https://guides.rubyonrails.org/active_model_basics.html#securepassword]
-# gem "bcrypt", "~> 3.1.7"
+gem "bcrypt", "~> 3.1.7"
```

Comment:
Why do we need both gems?

Reply:
1.byebug - using this gem we can control the program execution , you can track your data before byebug(you can check before the byebug you are getting your data or not)

2. bcrypt-  bcrypt is an hash algorithm, which safely stores our password(it stores in a hash value)

Reply:
1.byebug - using this gem we can control the program execution , you can track your data before byebug(you can check before the byebug you are getting your data or not)

2. bcrypt-  bcrypt is an hash algorithm, which safely stores our password(it stores in a hash value)